### PR TITLE
Reset model inputs after beam search work. Remove extra pipeline creation in tests

### PR DESCRIPTION
**Pull request by olpipi**

### CODE REVIEW

File: src/cpp/src/group_beam_searcher.cpp [hunk 1]
```diff
@@ -412,8 +412,14 @@ EncodedResults beam_search(ov::InferRequest& lm,
...
         size_t batch_size = next_tokens.size();
         // Set pointers
-        lm.set_tensor("input_ids", ov::Tensor{ov::element::i64, {batch_size, 1}, next_tokens.data()});
-        lm.set_tensor("beam_idx", ov::Tensor{ov::element::i32, {batch_size}, next_beams.data()});
+        ov::Tensor next_tokens_tensor(ov::element::i64, {batch_size, 1});
+        std::memcpy(next_tokens_tensor.data(), next_tokens.data(), next_tokens_tensor.get_byte_size());
+        lm.set_tensor("input_ids", next_tokens_tensor);
+
+        ov::Tensor next_beams_tensor(ov::element::i32, {batch_size});
+        std::memcpy(next_beams_tensor.data(), next_beams.data(), next_beams_tensor.get_byte_size());
```
Comment (ilya-lavrenov) [inline src/cpp/src/group_beam_searcher.cpp:420]: is it possible to reset `Tensor`s before exiting pipeline instead of copying data?

Comment (olpipi) [reply src/cpp/src/group_beam_searcher.cpp]: done.

Comment (olpipi) [reply src/cpp/src/group_beam_searcher.cpp]: Is it ok to make it only for beam search or should we reset inputs on pipeline level?

File: src/cpp/src/group_beam_searcher.cpp [hunk 2]
```diff
@@ -412,20 +419,18 @@ EncodedResults beam_search(ov::InferRequest& lm,
...
+        lm.set_tensor("beam_idx", ov::Tensor{ov::element::i32, {batch_size}, next_beams.data()});
 
         // Set auxiliary inputs
         update_attention_mask_with_beams(lm.get_tensor("attention_mask"), next_beams);
         if (position_ids_available)
             update_position_ids(lm.get_tensor("position_ids"), lm.get_tensor("attention_mask"));
     }
 
+    //reset all inputs with empty tensors
+    reset_inputs(lm);
```
Comment (Wovchena) [inline src/cpp/src/group_beam_searcher.cpp:432]: Rename the function to `reset_all_inputs_to_empty_tensors()` so you don't need the comment.

Comment (olpipi) [reply src/cpp/src/group_beam_searcher.cpp]: done

**Review by Wovchena — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**